openeo_processes_dask/process_implementations/cubes/_filter.py

```diff
@@ -15,9 +16,40 @@ def filter_bbox(data: RasterCube, extent: BoundingBox) -> RasterCube:
 
 
 def filter_temporal(
-    data: RasterCube, extent: TemporalInterval, dimension: str
+    data: RasterCube, extent: TemporalInterval, dimension: str = None
 ) -> RasterCube:
-    raise NotImplementedError()
+    temporal_dims = data.openeo.temporal_dims
+
+    if dimension is not None:
+        if dimension not in data.dims:
+            raise DimensionNotAvailable(
+                f"A dimension with the specified name: {dimension} does not exist."
+            )
+        applicable_temporal_dimension = dimension
+    else:
+        if not temporal_dims:
+            raise DimensionNotAvailable(
+                f"No temporal dimension detected on dataset. Available dimensions: {data.dims}"
+            )
+        if len(temporal_dims) > 1:
+            raise TooManyDimensions(
+                f"The data cube contains multiple temporal dimensions: {temporal_dims}. The parameter `dimension` must be specified."
+            )
+    applicable_temporal_dimension = temporal_dims[0]
+
+    start_time = extent[0]
+    if start_time is not None:
+        start_time = start_time.to_numpy()
```

Comment:
This is only possible if starttime is either `DateTime` or `Year`, since `Date` and `Time` are ambiguous. I'm not entirely sure whether this wants to be done here. Would it make sense to immediately disambiguate temporal intervals in the parser when possible and just always throw an error otherwise? 

I.e. if `start` is `2023-03-02T15:34:17+00:00` and `end` is `16:34:17+00:00`, assume that the same date is meant and disambiguate `end` to `2023-03-02T16:34:17+00:00`.
Otherwise we'd possibly be duplicating this disambiguation logic across multiple temporal processes.
wdyt?

Reply:
Indeed, if possible I would pass to dask-processes directly a numpy object, which would make it more portable and less dependant on the parser code.
For Date, I would just parse it like:
input: `2023-03-02` -> output: `2023-03-02T00:00:00+00:00` (check my PR in the parser, it's doing this)

Anyway, maybe we should firstly implement the simple scenarios (`DateTime`, where `Date` is also mapped to `DateTime`) and later on check if we can support `Time` only, which means supoprting `pendulum.Time` as an index variable in xArray.

Reply:
@LukeWeidenwalker how do we proceed here? Raise an error if we try to use a temporal class which is not supported at the moment?

Reply:
> Raise an error if we try to use a temporal class which is not supported at the moment?

Yeah, let's do it like this, we can extend this in the future!

Btw I've just pulled in the changes from your PR in the parser, so can use them here too!

Reply:
@LukeWeidenwalker I'll be on vacation until March 27. If you would like to close this PR before modify it as you prefer :)

Reply:
Ah, thanks for letting me know, I'll do that! - have a nice holiday 🌴